File: src/image.js
const Jimp = require('jimp');

function image() {

    let resolution
    let pngImage

    function create(res) {
        resolution = res
        pngImage = new Jimp(resolution, resolution)
        return this
    }

    function setPixelColor(color, x, y) {
        pngImage.setPixelColor(color, x, y)
        return this
    }

    function save(filePath) {
        pngImage.write(filePath)
        return this
    }  
    
    function fillBackgroundWith(color) {
        for(x = 0 ; x < resolution ; x++) {
            for(y = 0 ; y < resolution ; y++) {
                pngImage.setPixelColor(color, x, y);
            }
        }
        return this
    } 

    return {
        create,
        setPixelColor,
        save,
        fillBackgroundWith
    } 
}

module.exports = image();

File: mandelbrot/image.py
from PIL import Image as BaseImage


class Image:
    def __init__(self, resolution):
        self.img = BaseImage.new("RGB", (resolution, resolution), color="white")

    def set_pixel_color(self, x, y, color=(255, 255, 255)):
        self.img.putpixel((x, y), color)

    def save(self, file_path):
        self.img.save(file_path)
